ibis/selectors.py

```diff
@@ -362,12 +362,16 @@ def c(*names: str | ir.Column) -> Predicate:
     """Select specific column names."""
     names = frozenset(col if isinstance(col, str) else col.get_name() for col in names)
 
-    def func(col: ir.Value) -> bool:
-        schema = col.op().rel.schema
-        if extra_cols := (names - schema.keys()):
+    @functools.cache
+    def check_delta(schema):
+        if extra_cols := names - schema._name_locs.keys():
```

Comment:
`_name_locs` is slightly faster here because it's a computed `dict` whose methods are all implemented in C, compared to our `Schema` mapping which hits pure Python code paths for this operation.